medcat-v2/medcat/cat.py

```diff
@@ -834,6 +836,59 @@ def get_model_card(self, as_dict: bool = False) -> Union[str, ModelCard]:
             return model_card
         return json.dumps(model_card, indent=2, sort_keys=False)
 
+    @overload
```

Comment:
purpose of `@overload` here rather than `-> Union[str, ModelCard]` as you've done in the impl?